### ISSUE TITLE
Unable to upload image file using editor 

### ISSUE THREAD

**Issue by asudally** [bizley/yii2-content-tools#11]
```
    public function run()
    {
        if (!Yii::$app->request->isPost) {
            return Json::encode(['errors' => ['POST parameters are missing!']]);
        }
        try {
            $data = Yii::$app->request->post();
            if (empty($data['url'])) {
                throw new InvalidParamException('Parameter "url" is missing!');
            }
            $url = trim($data['url']);
            if (strpos($url, '/') === 0) {
                $url = substr($url, 1);
            }
            if (strpos($url, '?_ignore=') !== false) {
                $url = substr($url, 0, strpos($url, '?_ignore='));
            }
            $imageSizeInfo = @getimagesize($url);
            if ($imageSizeInfo === false) {
                throw new InvalidParamException('Parameter "url" seems to be invalid!');
            }
            if (!empty($data['crop'])) {
                $crop = explode(',', $data['crop']);
                if (count($crop) !== 4) {
                    throw new InvalidParamException('Parameter "crop" is invalid!');
                }
                $positions = [];
                foreach ($crop as $position) {
                    $position = trim($position);
                    if (!is_numeric($position) || $position < 0 || $position > 1) {
                        throw new InvalidParamException('Parameter "crop" contains invalid value!');
                    }
                    $positions[] = $position;
                }
                list($width, $height) = $imageSizeInfo;
                Image::crop(
                    $url,
                    floor($width * $positions[3] - $width * $positions[1]),
                    floor($height * $positions[2] - $height * $positions[0]),
                    [
                        floor($width * $positions[1]),
                        floor($height * $positions[0])
                    ]
                )->save($url);
            }
            return Json::encode([
                'size' => @getimagesize($url),
                'url' => '/' . $url,
                'alt' => basename($url)
            ]);
        } catch (Exception $e) {
            return Json::encode(['errors' => [$e->getMessage()]]);
        }
    }

```
Error    ```
 $imageSizeInfo = @getimagesize($url);
            if ($imageSizeInfo === false) {
                throw new InvalidParamException('Parameter "url" seems to be invalid!');
            }
```

**Comment by bizley** [comment]
Hmm, might be the same problem as in #10 ?

**Comment by bizley** [comment]
Sorry, I can not reproduce the error. Feel free to comment here if you believe this needs to be reopened (and you are willing to provide more info).

**Comment by BdeDev** [comment]
`http://localhost/yii2-base/site/content-tools-image-insert`
InsertAction $_POST data
```
Array
(
    [url] => /yii2-base/content-tools-uploads/1683.jpg
    [width] => 600
    [crop] => 0,0,1,1
    [_csrf] => CAJL3hv2USMWD9cy8LbbXzfcsYy-hEe7T68RFL5wNpV6ewWxb6AIbXFkv0iVzu0JdrP6z9nSF_p8_FBF0Dle5Q==
)
```

Error 
`{"errors":["Parameter \"url\" seems to be invalid!"]}`

**Comment by asudally** [comment]
`http://localhost/yii2-base/site/content-tools-image-insert`
InsertAction $_POST data
```
Array
(
    [url] => /yii2-base/content-tools-uploads/1683.jpg
    [width] => 600
    [crop] => 0,0,1,1
    [_csrf] => CAJL3hv2USMWD9cy8LbbXzfcsYy-hEe7T68RFL5wNpV6ewWxb6AIbXFkv0iVzu0JdrP6z9nSF_p8_FBF0Dle5Q==
)
```

Error 
`{"errors":["Parameter \"url\" seems to be invalid!"]}`

Method `getimagesize($url)` requires URL but then reaching to the below code we are doing this 

```
$url = trim($data['url']);

			if (strpos($url, '/') === 0)
			{
				$url = substr($url, 1);
			}
if (strpos($url, '?_ignore=') !== false)
			{
				$url = substr($url, 0, strpos($url, '?_ignore='));
			}

	$imageSizeInfo = @getimagesize($url);

			if ($imageSizeInfo === false)
			{
				throw new InvalidParamException('Parameter "url" seems to be invalid!');
			}
```

**Comment by bizley** [comment]
Ok, I see that problem occurs when it's run in localhost environment.

**Comment by asudally** [comment]
> Ok, I see that problem occurs when it's run in localhost environment.

Just for the confirmation same issue has been occurred on the live server as well .

`https://myDomain.com/myProject/site/content-tools-image-insert`
`{"errors":["Parameter \"url\" seems to be invalid!"]}
`

**Comment by bizley** [comment]
This should be fixed in [1.5.0](https://github.com/bizley/yii2-content-tools/releases/tag/1.5.0).

I've added the same options with `$uploadDir` and `$viewPath` for rotate and insert action as for upload action (added in 1.4.0) so now image while being processed is using server path instead of given web path. Because all the actions are standalone ones unfortunately you need to set these options to be the same in each action separately (of course only if you need them different than default).

Thanks for reporting this.
